app/actions/ControlActions.js

```diff
@@ -409,184 +265,111 @@ export const SETMAXPRICEABSOLUTE = "SETMAXPRICEABSOLUTE";
-    const { getTicketBuyerConfigResponse } = getState().control;
-    var hitError = "";
-    if (account != getTicketBuyerConfigResponse.getAccount()) {
-      var request = new SetAccountRequest();
-      request.setAccount(account);
-      ticketBuyerService.setAccount(request, function (error) {
-        if (error) {
-          hitError += error + ". ";
-        }
+export const setTicketBuyerConfigAttempt = (
+  account, balanceToMaintain, maxFee, maxPriceAbsolute, maxPriceRelative, stakePool, maxPerBlock
+) => (dispatch, getState) => {
+  const cfg = getWalletCfg();
+  const state = getState();
+  const ticketBuyerService = sel.ticketBuyerService(state);
+  const getTicketBuyerConfigResponse = sel.getTicketBuyerConfigResponse(state);
+  const promises = [];
+  dispatch({ type: SETTICKETBUYERCONFIG_ATTEMPT });
+
+  if (account !== getTicketBuyerConfigResponse.getAccount())
+    promises.push(wallet.setTicketBuyerAccount(ticketBuyerService, account));
+  if (balanceToMaintain*1e8 !== getTicketBuyerConfigResponse.getBalanceToMaintain())
+    promises.push(wallet
+      .setTicketBuyerBalanceToMaintain(ticketBuyerService, balanceToMaintain*1e8)
+      .then(() => {
+        cfg.set("balancetomaintain", balanceToMaintain);
+        dispatch({ balanceToMaintain, type: SETBALANCETOMAINTAIN });
+      }));
+  if (maxFee*1e8 !== getTicketBuyerConfigResponse.getMaxFee())
+    promises.push(wallet
+      .setTicketBuyerMaxFee(ticketBuyerService, maxFee*1e8)
+      .then(() => {
+        cfg.set("maxfee", maxFee);
+        dispatch({ maxFee, type: SETMAXFEE });
+      }));
+  if (maxPriceAbsolute*1e8 !== getTicketBuyerConfigResponse.getMaxPriceAbsolute())
+    promises.push(wallet
+      .setTicketBuyerMaxPriceAbsolute(ticketBuyerService, maxPriceAbsolute*1e8)
+      .then(() => {
+        cfg.set("maxpriceabsolute",maxPriceAbsolute);
+        dispatch({ maxPriceAbsolute, type: SETMAXPRICEABSOLUTE });
+      }));
+  if (maxPriceRelative !== getTicketBuyerConfigResponse().getMaxPriceRelative())
+    promises.push(wallet
+      .setTicketBuyerMaxPriceRelative(ticketBuyerService, maxPriceRelative)
+      .then(() => {
+        cfg.set("maxpricerelative",maxPriceRelative);
+        dispatch({ maxPriceRelative, type: SETMAXPRICERELATIVE });
+      }));
+  if (stakePool.TicketAddress !== getTicketBuyerConfigResponse.getVotingAddress())
+    promises.push(wallet.setTicketBuyerVotingAddress(ticketBuyerService, stakePool.TicketAddress));
+  if (stakePool.PoolAddress !== getTicketBuyerConfigResponse.getPoolAddress())
+    promises.push(wallet.setPoolAddress(ticketBuyerService, stakePool.PoolAddress));
+  if (stakePool.PoolFees !== getTicketBuyerConfigResponse.getPoolFees())
+    promises.push(wallet.setPoolFees(ticketBuyerService, stakePool.PoolFees));
+  if (maxPerBlock !== getTicketBuyerConfigResponse.getMaxPerBlock())
+    promises.push(wallet.setTicketBuyerMaxPerBlock(ticketBuyerService, maxPerBlock));
+
+  return promises
```

Comment:
This seems wrong to me... `promises` is an array, it shouldn't have the `.then()` method, but the function is craping out earlier than this when trying to update the config